test/parallel/test-cli-node-cli-manpage-env-vars.mjs

```diff
@@ -0,0 +1,105 @@
+import '../common/index.mjs';
+import assert from 'assert';
+import { createReadStream } from 'node:fs';
+import { createInterface } from 'node:readline';
+import { resolve, join } from 'node:path';
+
+// This test checks that all the environment variables defined in the public CLI documentation (doc/api/cli.md)
+// are also documented in the manpage file (doc/node.1) and vice-versa (that all the environment variables
+// in the manpage are present in the CLI documentation)
+
+const rootDir = resolve(import.meta.dirname, '..', '..');
+
+const cliMdEnvVarNames = await collectCliMdEnvVarNames();
+const manpageEnvVarNames = await collectManPageEnvVarNames();
+
+assert(cliMdEnvVarNames.size > 0,
+       'Unexpectedly not even a single env variable was detected when scanning the `doc/api/cli.md` file'
+);
+
+assert(manpageEnvVarNames.size > 0,
+       'Unexpectedly not even a single env variable was detected when scanning the `doc/node.1` file'
+);
+
+// TODO(dario-piotrowicz): add the missing env variables to the manpage and remove this set
+const knownEnvVariablesMissingFromManPage = new Set([
+  'NODE_COMPILE_CACHE',
+  'NODE_DISABLE_COMPILE_CACHE',
+  'NODE_PENDING_PIPE_INSTANCES',
+  'NODE_TEST_CONTEXT',
+  'NODE_USE_ENV_PROXY',
+]);
+
+for (const envVarName of cliMdEnvVarNames) {
+  if (!manpageEnvVarNames.has(envVarName) && !knownEnvVariablesMissingFromManPage.has(envVarName)) {
+    assert.fail(`The "${envVarName}" environment variable (present in \`doc/api/cli.md\`) is missing from the \`doc/node.1\` file`);
+  }
+  manpageEnvVarNames.delete(envVarName);
+}
+
+if (manpageEnvVarNames.size > 0) {
+  assert.fail(`The following env variables are present in the \`doc/node.1\` file but not in the \`doc/api/cli.md\` file: ${
+    [...manpageEnvVarNames].map((name) => `"${name}"`).join(', ')
+  }`);
+}
+
+async function collectManPageEnvVarNames() {
+  const manPagePath = join(rootDir, 'doc', 'node.1');
+  const fileStream = createReadStream(manPagePath);
+
+  const rl = createInterface({
+    input: fileStream,
+  });
+
+  const envVarNames = new Set();
+
+  for await (const line of rl) {
+    const match = line.match(/^\.It Ev (?<envName>[^ ]*)/);
+    if (match) {
+      envVarNames.add(match.groups.envName);
+    }
+  }
+
+  return envVarNames;
+}
+
+async function collectCliMdEnvVarNames() {
+  const cliMdPath = join(rootDir, 'doc', 'api', 'cli.md');
+  const fileStream = createReadStream(cliMdPath);
+
+  let insideEnvVariablesSection = false;
+
+  const rl = createInterface({
+    input: fileStream,
+  });
+
+  const envVariableRE = /^### `(?<varName>[^`]*?)(=[^`]+)?`$/;
```

Comment:
Nit:
```suggestion
  const envVariableRE = /^### `(?<varName>[^`]*?)(?:=[^`]+)?`$/;
```
Change to non-capturing group since we're not using the contents of this group.